article-analyzer/src/main/java/co/inventorsoft/academy/spring/notifier/NotificationManager.java

```diff
@@ -0,0 +1,31 @@
+package co.inventorsoft.academy.spring.notifier;
+
+import co.inventorsoft.academy.spring.model.User;
+import co.inventorsoft.academy.spring.notifier.impl.EmailNotificationService;
+import co.inventorsoft.academy.spring.notifier.impl.SlackNotificationService;
+import co.inventorsoft.academy.spring.service.UserService;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class NotificationManager {
+    private final UserService userService;
+    private final EmailNotificationService emailNotificationService;
+    private final SlackNotificationService slackNotificationService;
+
+    public void sendNotifications() {
+        List<User> users = userService.getUsers();
+        if (!users.isEmpty()) {
```

Comment:
remove this if at all, forEach will not be executed if list is empty, so no exception will happen

Reply:
fixed